src/lib/contract/ContractManager.js

```diff
@@ -0,0 +1,157 @@
+import ContractFactory from 'src/lib/contract/ContractFactory';
+import { ethers } from 'ethers';
+import axios from 'axios';
+import { ERC1155_TRANSFER_SIGNATURE } from 'src/lib/abi/signature/transfer_signatures.js';
+import { erc721MetadataAbi } from 'src/lib/abi';
+const tokenList = 'https://raw.githubusercontent.com/telosnetwork/token-list/main/telosevm.tokenlist.json';
+
+export default class ContractManager {
+    constructor(indexerApi, parser) {
+        this.contracts = {};
+        this.parser = parser;
+        this.factory = new ContractFactory();
+        this.indexerApi = indexerApi;
+        this.ethersProvider = new ethers.providers.JsonRpcProvider(process.env.NETWORK_EVM_RPC);
+    }
+
+    getEthersProvider() {
+        return this.ethersProvider;
+    }
+    async parseContractTransaction(data, contract) {
+        if (data === '0x' || data === null || typeof contract === 'undefined') {
+            return false;
+        }
+        if (contract.getInterface()) {
+            try {
+                return await contract.getInterface().parseTransaction({ data });
+            } catch (e) {
+                console.log(`Failed to parse transaction data ${data} using abi for ${contract.address}`);
```

Comment:
Not really because we have a fallback behind it, so getting here shouldn't change anything for the user just the way we get the data, which logs an error if it fails. Also it is expected we can reach that point in some cases (NFT that just got minted but not indexed yet)